docs/content/sips/013-sqlite.md

```diff
@@ -0,0 +1,165 @@
+  // The provided connection is not valid
+  invalid-connection,
+  // The database has reached its capacity
+  database-full,
+  // Some implementation-specific error has occurred (e.g. I/O)
+  io(string)
+}
+
+// Open a connection to a named database instance.
+//
+// If `database` is "default", the default instance is opened.
+//
+// `error::no-such-database` will be raised if the `name` is not recognized.
+open: func(name: string) -> expected<connection, error>
+
+// Execute a statement
+execute: func(conn: connection, statement: string, parameters: list<value>) -> expected<unit, error>
+
+// Query data
+query: func(conn: connection, query: string, parameters: list<value>) -> expected<query-result, error>
+
+// Close the specified `connection`.
+close: func(conn: connection)
+
+// A result of a query
+record query-result {
+  // The names of the columns retrieved in the query
+  columns: list<string>,
+  // The row results each containing the values for all the columns for a given row
+  rows: list<row-result>,
+}
+
+// A set of values for each of the columns in a query-result
+record row-result {
+  values: list<value>
+}
+
+// The values used in statements/queries and returned in query results
+variant value {
+  integer(s64),
+  real(float64),
+  text(string),
+  blob(list<u8>),
+  null
+}
+```
+
+*Note: the pseudo-resource design was inspired by the interface of similar functions in [WASI preview 2](https://github.com/bytecodealliance/preview2-prototyping/blob/d56b8977a2b700432d1f7f84656d542f1d8854b0/wit/wasi.wit#L772-L794).*
+
+#### Interface open questions
+
+* `row-result` can be very large. Should we provide some paging mechanism or a different API that allows for reading subsets of the returned data?
+  * Crossing the wit boundary could potentially be expensive if the results are large enough. Giving the user control of how they read that data could be helpful.
+* Is there really a need for query *and* execute functions since at the end of the day, they are basically equivalent?
+
+#### Database migrations
+
+Database tables typically require some sort of configuration in the form of database migrations to get table schemas into the correct state. To begin with a command line option supplied to `spin up` will be available for running SQL statements on start up (i.e., `--sqlite-migration "CREATE TABLE users..."`). It will be up to the user to provide itempotent statements such that running them multiple times does not produce unexpected results.
+
+However, such an approach (while useful) is likely to not be sufficient for more advanced use cases. We could require the user to ensure that the database is in the correct state each time the trigger handler function is run. However, there are a few issues with this:
+* Schema tracking schemes (e.g., a "migrations" table) themselves require some sort of bootstrap step.
+* This goes against the design principle of keeping components handler functions simple and single purpose.
```

Comment:
These two lines seem to be talking about something other than the CLI option... am I misunderstanding?

Reply:
Yes, this is about an alternative approach that we will *not* be doing. I'll try to make the wording more clear.